### PR TITLE
Fix AddPeering cmdlet for ipv6, add test for add peering cmdlet, refer 17.0.0.0-preview sdk version and generate new session recordings

### PR DESCRIPTION
**Pull request by vinayada1**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: src/ResourceManager/Network/Commands.Network/packages.config [hunk 1]
```diff
@@ -2,5 +2,8 @@
 <packages>
   <package id="AutoMapper" version="6.0.2" targetFramework="net452" />
   <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.Network" version="15.1.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.Network" version="17.0.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.10" targetFramework="net452" />
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network/packages.config]: These additional packages should not be here

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCircuitTests.ps1 [hunk 2]
```diff
@@ -42,9 +42,7 @@ function Test-ExpressRouteRouteFilters
       $resourceGroup = New-AzureRmResourceGroup -Name $rgname -Location $location
 
       # Create the route filter
-      $job = New-AzureRmRouteFilter -Name $filterName -ResourceGroupName $rgname -Location $location -Force -AsJob
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCircuitTests.ps1]: Why are you removing this?

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCircuitTests.ps1 [hunk 3]
```diff
@@ -100,10 +97,8 @@ function Test-ExpressRouteCircuitStageCRUD
       $resourceGroup = New-AzureRmResourceGroup -Name $rgname -Location $rglocation
       
       # Create the ExpressRouteCircuit
-	  $job = New-AzureRmExpressRouteCircuit -Name $circuitName -Location $location -ResourceGroupName $rgname -SkuTier Standard -SkuFamily MeteredData  -ServiceProviderName "equinix" -PeeringLocation "Silicon Valley" -BandwidthInMbps 500 -AllowClassicOperations $true -AsJob
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCircuitTests.ps1]: Please do not remove this

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCircuitTests.ps1 [hunk 4]
```diff
@@ -55,9 +53,7 @@ function Test-ExpressRouteRouteFilters
 	  $rule = New-AzureRmRouteFilterRuleConfig -Name $ruleName -Access Allow -RouteFilterRuleType Community -CommunityList "12076:5010" -Force
 	  $filter = Get-AzureRmRouteFilter -Name filter -ResourceGroupName filter
 	  $filter.Rules.Add($rule)
-	  $job = Set-AzureRmRouteFilter -RouteFilter $filter -Force -AsJob
```
Comment (markcowl) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCircuitTests.ps1]: Please do not remove the usage of jobs

**Comment by markcowl** [comment]
@vinayada1 and tests are still failing

File: src/ResourceManager/Network/Commands.Network/packages.config (hunk 1 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network/packages.config]: Resolved

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCircuitTests.ps1 (hunk 2 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCircuitTests.ps1]: Did not intend to change this. Not sure how these diffs got introduced. Resolved

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCircuitTests.ps1 (hunk 4 above)
Comment (vinayada1) [reply src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/ExpressRouteCircuitTests.ps1]: Resolved

File: src/ResourceManager/Network/Commands.Network/packages.config (hunk 1 above)
Comment (maddieclayton) [reply src/ResourceManager/Network/Commands.Network/packages.config]: Please remove these packages again.

**Comment by maddieclayton** [comment]
@vinayada1 One small change then this should be ready to merge.

**Comment by maddieclayton** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/417/

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/430/

**Comment by maddieclayton** [comment]
@vinayada1 These two tests still need rerecording: 
Commands.Network.Test.ScenarioTests.VirtualNetworkGatewayTests.TestSetVirtualNetworkGatewayCRUD
Commands.Network.Test.ScenarioTests.VirtualNetworkGatewayTests.VirtualNetworkGatewayRouteApiTest

**Comment by maddieclayton** [comment]
Rerun of on-demand here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/438/

**Comment by Nilambari** [comment]
@maddieclayton Recorded failing tests and build is also passing. Can you help merge this PR. Thanks!

**Comment by maddieclayton** [comment]
Pushing into preview now: this will be included in the next release.

**Pull request merged**